### PR TITLE
Special patterns of constraint to penalty converters

### PR DESCRIPTION
**Pull request by knamba-jp**
### Summary
Resolves https://github.com/Qiskit/qiskit-optimization/issues/40

TODO
- [x] Implementation of the converter
- [x] Intuitive converter name
- [x] Docstring
- [x] Unit test

### Details and comments
Owned by Amano-san and me

[A Tutorial on Formulating and Using QUBO Models](https://arxiv.org/abs/1811.11538) introduces examples of the special patterns in page 10 as follows.
image

### CODE REVIEW

**Comment by t-imamichi** [comment]
We need a more intuitive name than `SpecialConstraintToPenalty` because there might be other types of inequality to equality converters in the future.

**Comment by t-imamichi** [comment]
Could you apply black code formatter by `make black` as explained in https://github.com/Qiskit/qiskit-optimization/blob/main/CONTRIBUTING.md#pull-request-checklist
We recently apply black to the entire code base. See #117 for details.

**Comment by ibmamnt** [comment]
We will run black command as suggested. As for the name of the class.
@t-imamichi How about this name ?

  KnownConstraintConverter

**Comment by t-imamichi** [comment]
Thanks. The code format looks good. The naming is a bit difficult. What is the best name for this converter...? Maybe we can leave it later.

**Comment by t-imamichi** [comment]
`LinearInequalityToPenalty` may work.

FYI, you don't have to implement patterns `x+y=1` and `x=y` because `LinearEqualityToPenalty` can handle them.

File: test/converters/test_converters.py [hunk 1]
```diff
@@ -23,23 +23,27 @@
...
+    InequalityToEquality,
+    IntegerToBinary,
+    LinearEqualityToPenalty,
+    LinearInequalityToPenalty,
+)
 from qiskit_optimization.problems import Constraint, Variable
 
 logger = logging.getLogger(__name__)
 
 QUBIT_OP_MAXIMIZE_SAMPLE = (
```
Comment (t-imamichi) [inline test/converters/test_converters.py]: You can simplify it by using `PauliSumOp.from_list`.
https://qiskit.org/documentation/stubs/qiskit.opflow.primitive_ops.PauliSumOp.html?highlight=paulisumop#qiskit.opflow.primitive_ops.PauliSumOp.from_list

Comment (knamba-jp) [reply test/converters/test_converters.py]: We discussed and concluded it should be done in another PR.

**Comment by knamba-jp** [comment]
Since some errors can't be fixed, we'll open another PR.

**Pull request closed without merging**